### PR TITLE
Write Builders manually 2

### PR DESCRIPTION
**Pull request by burrbull**
#101  without changes in Error hierarchy.

Should slightly be easier to review.

### CODE REVIEW

**Comment by burrbull** [comment]
bors try

File: src/lib.rs [hunk 1]
```diff
@@ -92,6 +92,10 @@ pub(crate) fn new_element(name: &str, text: Option<String>) -> Element {
     }
 }
 
+pub trait Build {
+    type Builder;
+}
+
```
Comment (Emilgardis) [inline src/lib.rs:98]: Why is this not

```suggestion
pub trait Builder {
    type Output;
    type Error: std::error::Error + std::fmt::Display + 'static; // Maybe Send + Sync even?
    pub fn build(self) -> Result<Self::Output, Self::Error>;
}
```
and then move every impl of fn build to the trait. I don't see the reason for having this trait unless it is used like this.

Comment (burrbull) [reply src/lib.rs]: I removed this trait as it is not important. We can discuss this in future.

**Comment by burrbull** [comment]
bors try

**Comment by burrbull** [comment]
Done.

**Pull request merged**